WeakAuras/GenericTrigger.lua

```diff
@@ -639,9 +639,10 @@ local function RunTriggerFunc(allStates, data, id, triggernum, event, arg1, arg2
       else
         ok, returnValue = xpcall(data.triggerFunc, errorHandler, allStates, event, arg1, arg2, ...);
       end
-      if (ok and returnValue) then
+      if (ok and (returnValue or (returnValue ~= false and allStates.__changed))) then
```

Comment:
it is perhaps a tiny bit weird that code which doesn't use the utility functions can set `__changed = true` but on balance i think it's fine